linera-execution/src/system.rs

```diff
@@ -517,9 +517,31 @@ where
                 for (chain_id, stream_id, next_index) in streams {
                     let subscriptions = self
                         .event_subscriptions
-                        .get_mut_or_default(&(chain_id, stream_id))
+                        .get_mut_or_default(&(chain_id, stream_id.clone()))
+                        .await?;
+                    if subscriptions.next_index >= next_index {
```

Comment:
How could it happen that we have an `UpdateStream` call that tries to consume an event with index lower than what it is currently? It'd mean that that this event stream has already been updated by some other block proposer on this chain?

Reply:
Yes, exactly. It's basically an error on the client side. We could disallow it, or just take the maximum.